File: js/spinWords.js
function spinWords(word) {
  let results = [];
  // turn string of words into array of words
  word = word.split(" ");
  word.forEach(word => {
    // if length of word is more than 5
    if (word.length > 4) {
      // split word into individual character array,
      // reverse the whole array and join back to a string
      results.push(
        word
          .split("")
          .reverse()
          .join(""),
      );
    } else {
      results.push(word);
    }
  });
  // return results array back into string form
  return results.join(" ");
}

File: python/spin_words.py
def spin_words(sentence):
    result = []
    # turn string into list
    new_sentence = sentence.split()
    # iterate over each word
    for word in new_sentence:
        # if greater than 4, reverse
        if len(word) >= 5:
            # [start:end:count] so this reverses the word
            result.append(word[::-1])
        else:
            result.append(word)
    # turn back into string w/ space between each item in list
    return " ".join(result)
